tests/subsys/llext/simple/src/test_llext_simple.c

```diff
@@ -268,6 +268,53 @@ static LLEXT_CONST uint8_t multi_file_ext[] ELF_ALIGN = {
 LLEXT_LOAD_UNLOAD(multi_file, true, NULL)
 #endif
 
+static LLEXT_CONST uint8_t export_dependent_ext[] ELF_ALIGN = {
+	#include "export_dependent.inc"
+};
+
+ZTEST(llext, test_inter_ext)
+{
+	void *dependency_buf = k_heap_alloc(&tllext_heap, sizeof(hello_world_ext), K_NO_WAIT);
+
+	zassert_not_null(dependency_buf);
+
+	void *dependent_buf = k_heap_alloc(&tllext_heap, sizeof(export_dependent_ext), K_NO_WAIT);
+
+	zassert_not_null(dependent_buf);
+
+	memcpy(dependency_buf, hello_world_ext, sizeof(hello_world_ext));
+	memcpy(dependent_buf, export_dependent_ext, sizeof(export_dependent_ext));
```

Comment:
Why do you need to copy the `_ext` buffers?

Reply:
there's a comment in the respective commit - because I reuse "hello world" and I thought it could be done by others from time to time too. And if we relocate extensions in place, that breaks

Reply:
@pillo79 https://github.com/zephyrproject-rtos/zephyr/pull/77738/commits/47f0f1c9174beb492f5456a1a4748e44db9f9b55

Reply:
Sorry, it was a quick review and I did not check the commits. :man_facepalming: 
However there are two things I am not fond of: 
1) it is being applied even when `STORAGE_WRITABLE` is not enabled. This makes the LLEXT code always access ELF data in RAM, and thus will no longer catch bugs that cause LLEXT to write something there. 
2) even if applied only on `STORAGE_WRITABLE`, this means that ELF buffers are never written to - they should be made fully `const` (not `LLEXT_CONST`).